src/page/Page.tsx

```diff
@@ -125,7 +125,9 @@ export const NeoPage = ({
   const [lastElement, setLastElement] = React.useState(<div key={getReportKey(pagenumber, '999999')}></div>);
   const [animated, setAnimated] = React.useState(false); // To turn off animations when cards are dragged around.
   const [isListOpen, setListOpen] = React.useState(true);
-
+  const enableSaveButtonForIds = reports.filter(
+    (report: any) => report.settings.hideQueryEditorInAutoRunOnMode === 'on'
```

Comment:
What if settings has no hideQueryEditorInAutoRunMode in the JSON? Will this access raise null error?

Reply:
Added optional chaining condition to handle any runtime errors. Thanks for pointing out